include/libtorrent/sha1_hash.hpp

```diff
@@ -161,7 +161,7 @@ namespace aux {
 		}
 		bool operator!=(digest32 const& n) const
 		{
-			return !std::equal(n.m_number, n.m_number + number_size, m_number);
```

Comment:
I'm on-board with making the operators non-members, and to make them all ``noexcept``.
Specifically what I'm questioning is to always express ``!=`` in terms of ``==``. Obviously the semantics should be the opposite, but you're not changing the semantics, just how it's expressed.